### PR TITLE
Uses an enum class for the Mgmt OpType

### PR DESCRIPTION
**Pull request by zwoop**
This fixes an issue where the behavior of va_start is undefined when using
the enum as the <n>'th argument (before the ...). Switching to the enum class
fixes this issue, and is also nicer (as per previous enum class changes).

### CODE REVIEW

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/135/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/149/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/132/

File: mgmt/api/NetworkMessage.h [hunk 1]
```diff
@@ -61,9 +61,9 @@ typedef enum {
   RECORD_DESCRIBE_CONFIG,
   LIFECYCLE_MESSAGE,
   UNDEFINED_OP /* This must be last */
-} OpType;
+};
 
-#define MGMT_OPERATION_TYPE_MAX (UNDEFINED_OP)
+#define MGMT_OPERATION_TYPE_MAX static_cast<unsigned>(OpType::UNDEFINED_OP)
```
Comment (bryancall) [inline mgmt/api/NetworkMessage.h]: A const OpType would be better than macro here.

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1831/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1722/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/261/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/393/

File: mgmt/api/NetworkMessage.h (hunk 1 above)
Comment (zwoop) [inline mgmt/api/NetworkMessage.h]: Changed it to a static_assert as per amc's recommendation. Then bike shedding ensued, so ignoring him now.

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/136/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/150/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/133/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1832/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/262/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1723/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/394/

**Comment by atsci** [comment]
RAT check *failed*! https://ci.trafficserver.apache.org/job/RAT-github/154/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/140/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/155/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/137/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/265/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1836/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/138/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1727/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/266/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/397/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/398/

**Pull request merged**